### DISCUSSION TITLE
Can agency swarm genesis be used with Azure OpenAI

### DISCUSSION THREAD

**Discussion by nuvoqueonline** [VRSEN/agency-swarm#102] [Q&A]
Can agency swarm genesis be used with Azure OpenAI? I've tried changing the get_openai_client() however the change isn't working as the OpenAI key is referenced in multiple files. is there a specific file that must be modifed to change to azure?

**Comment by nagendrakumar02** [comment]
I am having the same question. Please let me know if you were able to find an answer to this question.

**Comment by mikkel-rs** [comment]
Yes, it can. Azure also has the Assistants API.

**Reply by ZanzyTHEbar** [reply]
~~That's awesome, but how. I see no method to override the URL in the source code.~~
I went through the source-code line by line, and i was wrong. I see that we can `set_openai_client` at any point and this will override the default client. Due to this, we can customize the `OpenAI` client and handle Azure or even other proxies.

**Comment by dhawalkgupta90** [comment]
Can you please suggest the changes? As far as I see, there's a lot of references to openai class and just moving the client to be AzureOpenAI would not be sufficient

**Comment by dabrodev** [comment]
Is there Azure OpenAI integration already available?

**Comment by nicko-ai** [comment]
Hey everyone! I hope this helps:
- https://vrsen.github.io/agency-swarm/advanced-usage/azure-openai/
